presto-native-execution/src/test/java/com/facebook/presto/spark/TestPrestoSparkNativeExecution.java

```diff
@@ -62,45 +65,53 @@
+                .setCatalogSessionProperty("hive", PUSHDOWN_FILTER_ENABLED, "true");
+
         if (System.getProperty("NATIVE_PORT") == null) {
             sessionBuilder.setSystemProperty(NATIVE_EXECUTION_EXECUTABLE_PATH, requireNonNull(System.getProperty("PRESTO_SERVER"), "Native worker binary path is missing. Add -DPRESTO_SERVER=/path/to/native/process/bin to your JVM arguments."));
         }
         return sessionBuilder.build();
     }
 
-    @Override
-    protected QueryRunner createQueryRunner()
-    {
-        Map<String, String> configs = new HashMap<>();
-        // prevent to use the default Prestissimo config files since the Presto-Spark will generate the configs on-the-fly.
-        configs.put("catalog.config-dir", "/");
-        return PrestoSparkQueryRunner.createHivePrestoSparkQueryRunner(configs);
-    }
-
-    @Test
-    public void testNativeExecutionWithProjection()
+    protected void assertQuery(String sql)
     {
-        assertUpdate("CREATE TABLE test_order WITH (format = 'DWRF') as SELECT orderkey, custkey FROM orders LIMIT 100", 100);
-
-        Session session = Session.builder(getNativeSession())
-                .setSystemProperty("table_writer_merge_operator_enabled", "false")
-                .setCatalogSessionProperty("hive", "collect_column_statistics_on_write", "false")
-                .setCatalogSessionProperty("hive", PUSHDOWN_FILTER_ENABLED, "true")
-                .build();
-
         // Reset the spark context to register the native execution shuffle manager. We want to let the query runner use the default spark shuffle
         // manager to generate the test tables and only test the new native execution shuffle manager on the test below test cases.
         PrestoSparkQueryRunner queryRunner = (PrestoSparkQueryRunner) getQueryRunner();
         queryRunner.resetSparkContext(getNativeExecutionShuffleConfigs());
         try {
-            assertQuerySucceeds(session, "SELECT * FROM test_order");
+            assertQuery(getNativeSession(), sql, TEST_SESSION, sql);
         }
         finally {
             queryRunner.resetSparkContext();
         }
     }
 
+    @Override
+    protected QueryRunner createQueryRunner()
+    {
+        // prevent to use the default Prestissimo config files since the Presto-Spark will generate the configs on-the-fly.
+        return PrestoSparkQueryRunner.createHivePrestoSparkQueryRunner(ImmutableMap.of("catalog.config-dir", "/"), ImmutableMap.of("hive.storage-format", "DWRF"));
+    }
+
+    @Override
+    protected ExpectedQueryRunner createExpectedQueryRunner()
+            throws Exception
+    {
+        LocalQueryRunner runner = new LocalQueryRunner(TEST_SESSION);
+        runner.createCatalog(TEST_SESSION.getCatalog().get(), new TpchConnectorFactory(), ImmutableMap.of());
+        return runner;
+    }
+
+    @Test
+    public void testNativeExecutionWithProjection()
+    {
+        assertQuery("SELECT orderkey, custkey FROM orders WHERE orderkey <= 200");
```

Comment:
Let's add a few queries here to make sure it works properly.

Reply:
Right now queries in the test suite also suffer from the multiple split per operator issue (#19201), I'd prefer to add more queries in the test after #19201 merged to avoid dramatically slow down the CI (it takes 10min to finish one query on my laptop).

Reply:
@miaoever 10 min for such a simple query is way too long. I see that #19201 is merged now. Would you rebase and check how long the query takes?